pkg/controllers/management/rkeworkerupgrader/upgrade.go

```diff
@@ -310,8 +310,15 @@ func (uh *upgradeHandler) upgradeCluster(cluster *v3.Cluster, nodeName string, p
 
 	notReady := len(status.notReady)
 	if notReady > maxAllowed {
-		return fmt.Errorf("cluster [%s] worker-upgrade: not enough nodes to upgrade: nodes %v notReady %v maxUnavailable %v",
-			clusterName, status.filtered, keys(status.notReady), maxAllowed)
+		v32.ClusterConditionUpgraded.False(cluster)
+		v32.ClusterConditionUpgraded.Message(cluster, fmt.Sprintf("cluster [%s] worker-upgrade: not enough nodes to upgrade: nodes %v notReady %v maxUnavailable %v",
+			clusterName, status.filtered, keys(status.notReady), maxAllowed))
+
+		if _, err := uh.clusters.Update(cluster); err != nil {
+			return err
+		}
+
+		return fmt.Errorf(v32.ClusterConditionUpgraded.GetStatus(cluster))
```

Comment:
I think `GetStatus` returns `True`, `False`, `Unknown`.
```suggestion
		return fmt.Errorf(v32.ClusterConditionUpgraded.GetMessage(cluster))
```